src/System.Collections.Immutable/ref/System.Collections.Immutable.cs

```diff
@@ -249,7 +249,7 @@ public sealed partial class Builder : System.Collections.Generic.ICollection<T>,
         }
         public partial struct Enumerator
         {
-            private readonly T[] _array;
+            private T[] _array;
```

Comment:
I don't know why auto-gen removes the readonly from here. Is this expected?

https://github.com/dotnet/corefx/blob/16f3eef83d6df4c9dbc0e2992dafbc9553db363d/src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.Enumerator.cs#L24

Reply:
Presumably because it has no impact on the semantics of the ref?

Reply:
Is this case different because this is a private field? I recall previously (for public properties), we wanted to retain the `readonly` keyword.

Reply:
I can take a look with you tomorrow on why it dropped it. Might be my fault, but not quite sure.

Reply:
> Is this case different because this is a private field? 

Yes. If it were public, readonly impacts things like whether the compiler needs to make a defensive copy when using the field in certain ways.

Reply:
I highlighted this because auto-gen kept the readonly keyword for `ArraySegment<T>` (within System.Runtime ref):
https://github.com/dotnet/corefx/blob/e6a5951fb68a09598f1fc1826a94de212385c649/src/Common/src/CoreLib/System/ArraySegment.cs#L35

But dropped it in its Enumerator:
https://github.com/dotnet/corefx/blob/e6a5951fb68a09598f1fc1826a94de212385c649/src/Common/src/CoreLib/System/ArraySegment.cs#L316

https://github.com/dotnet/corefx/blob/e6a5951fb68a09598f1fc1826a94de212385c649/src/System.Runtime/ref/System.Runtime.cs#L293-L332

![image](https://user-images.githubusercontent.com/6527137/61738808-f36c0b80-ad3f-11e9-91c8-ea78c7342941.png)

Looks like this discrepancy is related to your change to support nullability, @safern:

Reply:
Reverting this change for now.